src/api/app/models/project.rb

```diff
@@ -58,7 +58,10 @@ def autocomplete(search)
   has_many :messages, :as => :db_object, :dependent => :delete_all
   has_many :watched_projects, :dependent => :destroy, inverse_of: :project
 
-  has_many :linkedprojects, -> { order(:position) }, :class_name => 'LinkedProject', foreign_key: :db_project_id, :dependent => :delete_all
+  has_many :linked_projects, -> { order(:position) }, foreign_key: :db_project_id, :dependent => :delete_all
+  has_many :projects_linked, through: :linked_projects, class_name: 'Project', source: :linked_db_project
+  has_many :linking_projects, -> { order(:position) }, :class_name => 'LinkedProject', foreign_key: :linked_db_project_id, :dependent => :delete_all
+  has_many :projects_linking, through: :linking_projects, class_name: 'Project', source: :project
```

Comment:
Yes, I have the same feeling..., `linked_projects` is just the collection of the model in the middle that links two projects, so I created `projects_linked` that are the Project models linked from one project. We can rename that as: `links` for the actual linked_projects and `linked_projects` will be the Project models linked...
In the other hand we need the inverse association, so we can rename `linking_projects` for linking and projects_linking can stay as it is.

That's a naming mess! :smile: